client/settings/fraud-protection/interfaces.ts

```diff
@@ -0,0 +1,63 @@
+/**
+ * External dependencies
+ */
+import { Dispatch, SetStateAction } from 'react';
+
+export interface FraudPreventionSetting {
+	block: boolean;
+	enabled: boolean;
+	min_items?: string | number | null;
+	max_items?: string | number | null;
+	min_amount?: string | number | null;
+	max_amount?: string | number | null;
+}
+
+export type ProtectionSettingsUI = Record< string, FraudPreventionSetting >;
+
+export interface FraudPreventionSettingsContextType {
+	protectionSettingsUI: ProtectionSettingsUI;
+	setProtectionSettingsUI: ( settings: ProtectionSettingsUI ) => void;
+	protectionSettingsChanged: boolean;
+	setProtectionSettingsChanged: Dispatch< SetStateAction< boolean > >;
+}
+
+export interface FraudProtectionSettingsSingleCheck {
+	key: string;
+	value: any;
+	operator: string;
+}
+
+export interface FraudProtectionSettingsMultipleChecks {
+	operator: string;
+	checks: FraudProtectionSettingsSingleCheck[];
+}
+
+export type FraudProtectionSettingsCheck =
+	| FraudProtectionSettingsSingleCheck
+	| FraudProtectionSettingsMultipleChecks
+	| null;
+
+export interface AdvancedFraudProtectionSettings {
```

Comment:
This should be a definition for a `FraudProtectionRule` object. `AdvancedFraudProtectionSettings` is an array of `FraudProtectionRule`.